src/common/gui/CPatchBrowser.cpp

```diff
@@ -155,8 +157,10 @@ void CPatchBrowser::populatePatchMenuForCategory( int c, COptionMenu *contextMen
         
         if (!single_category)
         {
-            contextMenu->addEntry(subMenu, name);
-            subMenu->forget(); // Important, so that the refcounter gets it right
+            CMenuItem *entry = contextMenu->addEntry(subMenu, name);
+      		if (c == current_category)
+				entry->setChecked(true);
+           subMenu->forget(); // Important, so that the refcounter gets it right
```

Comment:
The indentation here is off by one space in on the subMenu and looks like it has hard tabs on the if.